Sources/Algorithms/Trim.swift

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension BidirectionalCollection {
+
+  /// Returns a `SubSequence` formed by discarding all elements at the start and end of the Collection
+  /// which satisfy the given predicate.
+  ///
+  /// Trimming is a common operation in computing, often referring to removing leading and trailing whitespace
+  /// from a string[1].
```

Comment:
I don't think we need this explanation. Let's introduce the example, instead — something like:

> This example uses `trimmed(where:)` to get a substring without the white space at the beginning and end of the string.